centreon/www/class/config-generate/abstract/host.class.php

```diff
@@ -210,14 +214,27 @@ protected function getMacros(&$host)
 
         $host['macros'] = array();
         foreach ($macros as $macro) {
-            $host['macros'][preg_replace(
+            $hostMacroName = preg_replace(
                 '/\$_HOST(.*)\$/',
                 '_$1',
-                $macro['host_macro_name']
-            )] = $macro['host_macro_value'];
+                $macro['host_macro_name']);
+
+            # Modify macro value for Centreon Vault Storage
+            if (
+                $this->isVaultEnabled
+                && $macro['is_password'] === 1
+            ) {
+                $macro['host_macro_value'] = sprintf("{%s::%s}", $hostMacroName, $macro['host_macro_value']);
+            }
+
+            $host['macros'][$hostMacroName] = $macro['host_macro_value'];
         }
         if (!is_null($host['host_snmp_community']) && $host['host_snmp_community'] != '') {
-            $host['macros']['_SNMPCOMMUNITY'] = $host['host_snmp_community'];
+            if (preg_match(self::VAULT_PATH_REGEX, $host['host_snmp_community'])) {
+                $host['macros']['_SNMPCOMMUNITY'] = sprintf("{%s::%s}", "_SNMPCOMMUNITY", $host['host_snmp_community']);
```

Comment:
check that macro name is valid, as it has been defined as "_HOSTSNMPCOMMUNITY" into vault